lumen/views/base.py

```diff
@@ -89,18 +98,22 @@ def _table_data(self) -> IO:
             io = BytesIO()
         data = self.view.get_data()
         if self.format == 'csv':
-            data.to_csv(io, **self.kwargs)
+            data.to_csv(io, index=self.index, **self.kwargs)
         elif self.format == 'json':
-            data.to_json(io, **self.kwargs)
+            data.to_json(io, index=self.index, **self.kwargs)
         elif self.format == 'xlsx':
-            data.to_excel(io, **self.kwargs)
+            data.to_excel(io, index=self.index, **self.kwargs)
         elif self.format == 'parquet':
-            data.to_parquet(io, **self.kwargs)
+            data.to_parquet(io, index=self.index, **self.kwargs)
         io.seek(0)
         return io
 
     def __panel__(self) -> DownloadButton:
-        filename = f'{self.view.pipeline.table}.{self.format}'
+        if self.filename:
+            filename = self.filename
+        else:
+            filename = slugify(self.view.title or self.view.pipeline.table)
```

Comment:
Not sure I'd expect it to use the title here. I get that it's a presumably unique label but it could also be quite long and unsuited to usage as a filename.

Reply:
It is more descriptive than `self.view.pipeline.table`, especially when multiple branched pipelines are getting the same base data with the same name. This means that all downloaded files will have the same name even if they come from different views.

Basically, I'm trying to avoid confusion about whether the downloaded data is raw or filtered/transformed. I'm open to suggestions; I agree it is not the most elegant way to do it.

Reply:
Very much get that point but I think adding the`filename` parameter offers a very explicit escape hatch. The 3 way fallback chain is difficult to explain and can result in pretty silly/garbled filenames.

Reply:
I've got a suggestion and feel free to shoot it down, how about allowing providing both the filename and extension when setting up the download functionality, i.e. setting `download: my_custom_filename.csv` would get parsed out into the format and the filename.

Reply:
Why would I shoot down such a good idea :+1: 

Added that in [369ae81](https://github.com/holoviz/lumen/pull/438/commits/369ae81906afff461cb10d6fe68e470cbb63e91c).